### ISSUE TITLE
Period/DatetimeIndex may not be broadcasting correctly

### ISSUE THREAD

**Issue by jtratner** [pydata/pandas#5032]
First two examples pass, third  example doesn't.
```python
# GH 4629
# arithmetic datetime64 ops with an index
s = Series(date_range('20130101',periods=5),index=date_range('20130101',periods=5))
expected = s-s.index.to_series()
result = s-s.index
assert_series_equal(result,expected)

result = s-s.index.to_period()
assert_series_equal(result,expected)

result2 = s.index.to_period() - s
expected2 = s.index.to_series() - s
assert_series_equal(result2, expected2)
```

Traceback:

```python
  /usr/local/bin/vim +2583 pandas/tests/test_series.py  # test_datetime64_with_index
    result2 = s.index.to_period() - s
  /usr/local/bin/vim +873  pandas/tseries/period.py  # __sub__
    return PeriodIndex(ordinal=self.values - other, freq=self.freq)
TypeError: ufunc subtract cannot use operands with types dtype('int64') and dtype('<M8[ns]')
```

**Comment by maximilianr** [comment]
@jreback this now works fine for me:
```
In [47]: expected = s-s.index.to_series()

In [48]: result = s-s.index

In [49]: assert_series_equal(result,expected)

In [50]: result2 = s.index.to_period() - s

In [51]: expected2 = s.index.to_series() - s

In [52]: assert_series_equal(result2, expected2)

In [53]: expected = s-s.index.to_series()

In [54]: result = s-s.index

In [55]: assert_series_equal(result,expected)

In [56]: 

In [56]: result = s-s.index.to_period()

In [57]: assert_series_equal(result,expected)

In [58]: 

In [58]: result2 = s.index.to_period() - s

In [59]: expected2 = s.index.to_series() - s

In [60]: assert_series_equal(result2, expected2)

In [61]: 
```

**Comment by jreback** [comment]
hmm must have been a duplicate issue 
closed by #10568
